rfcs/0002-authentication.md

```diff
@@ -1,12 +1,12 @@
 ---
-RFC: 0002
+RFC: 2
```

Comment:
I like this. I think we should keep the `0002` in the filenames like you have it for sorting purposes, but this does help in the actual document where it's not needed.